Description: move changes in source files to patches
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 03 Mar 2012
Forwarded: patch is from upstream

Index: astronomical-almanach/kfiles.c
===================================================================
--- astronomical-almanach.orig/kfiles.c	2012-03-02 12:32:19.000000000 +0100
+++ astronomical-almanach/kfiles.c	2012-03-02 12:32:40.000000000 +0100
@@ -3,7 +3,7 @@
  * or file containing orbital elements.
  */
 
-#if __BORLANDC__
+#if __BORLANDC__ || __STDC__
 #include <stdlib.h>
 #endif
 
@@ -26,8 +26,8 @@
 
 extern char *intfmt, *strfmt;/* see dms.c */
 
-static char starnam[80] = {'s','t','a','r','.','c','a','t','\0'};
-static char orbnam[80] = {'o','r','b','i','t','.','c','a','t','\0'};
+static char starnam[80] = "/usr/share/aa/star.cat";
+static char orbnam[80] = "/usr/share/aa/orbit.cat";
 static int linenum = 1;
 
 /* Read initialization file aa.ini
@@ -73,8 +73,19 @@
 printf( "\n\tSteve Moshier's Ephemeris Program v5.6\n\n" );
 printf( "Planetary and lunar positions approximate DE404.\n" );
 
-f = fopen( "aa.ini", "r" );
-if( f )
+{
+   char *t = getenv("HOME");
+   strcpy(s, "aa.ini");
+   if (t && strlen(t)<70) 
+     {
+       strcpy(s,t);
+       strcat(s,"/.aa.ini");
+     }
+}
+
+if( (f=fopen("aa.ini","r")) ||
+    (f=fopen(s,"r")) ||
+    (f=fopen("/etc/aa.ini","r")) )
 	{
 	fgets( s, 80, f );
 	sscanf( s, "%lf", &tlong );
