Description: move changes in source files to patches
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 03 Mar 2012
Forwarded: patch is from upstream

Index: astronomical-almanach/mar404.c
===================================================================
--- astronomical-almanach.orig/mar404.c	2012-03-02 12:33:32.000000000 +0100
+++ astronomical-almanach/mar404.c	2012-03-02 12:33:59.000000000 +0100
@@ -1344,7 +1344,7 @@
 
 };
 
-static char FAR args[] = {
+static char SIGNED FAR args[] = {
   0,  4,
   3,  4,  3, -8,  4,  3,  5,  2,
   3,  5,  2, -6,  3, -4,  4,  0,
