Description: move changes in source files to patches
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 03 Mar 2012
Forwarded: patch is from upstream

Index: astronomical-almanach/nep404.c
===================================================================
--- astronomical-almanach.orig/nep404.c	2012-03-02 13:17:16.000000000 +0100
+++ astronomical-almanach/nep404.c	2012-03-02 13:17:36.000000000 +0100
@@ -386,7 +386,7 @@
 
 };
 
-static char FAR args[] = {
+static char SIGNED FAR args[] = {
   0,  3,
   2,  1,  7, -2,  8,  0,
   3,  3,  5, -8,  6,  3,  8,  0,
